frontend/src/modules/controllers/ValueModal.tsx

```diff
@@ -105,167 +67,7 @@ interface ContentProps {
-   * Triggers callback when modal status changes
-   */
-  useEffect(() => {
-    if (updateModalStatus) {
-      updateModalStatus(open);
-    }
-  });
-
-  /**
-   * Function for handling opening of the model.
-   */
-  const handleOpen = () => {
-    setOpen(true);
-  };
-  /**
-   * Function for handling closing of the model.
-   */
-  const handleClose = () => {
-    setOpen(false);
-  };
-  /**
-   * Function for handling confirmation of the model.
-   */
-  const handleConfirm = () => {
-    requestCommitSetting(value);
-    setOpen(false);
-  };
-
-  /**
-   * Function for handling label changed for PIP.
-   *
-   * @param {string} label - Value label
-   *
-   * @returns string
-   */
-  function pipClarify(label: string) {
-    if (label === 'PIP') return '*not PEEP compensated';
-    return '';
-  }
-
-  const modalContent = (
-    <Grid container direction="row">
-      <Grid container item xs direction="column" className={classes.contentContainer}>
-        <Grid item>
-          <Typography variant="h4">
-            {label}
-            <Typography variant="h6" style={{ opacity: 0.8 }}>
-              {pipClarify(label)}
-            </Typography>
-          </Typography>
-        </Grid>
-        <Grid container item xs wrap="nowrap">
-          <Grid container item alignItems="baseline">
-            <Typography align="left" style={{ fontSize: '9.5rem' }}>
-              {value.toFixed(0)}
-            </Typography>
-            <Typography align="center" variant="h5" className={classes.unitsLabel}>
-              {units}
-            </Typography>
-          </Grid>
-        </Grid>
-      </Grid>
-      <Grid item>
-        <ValueClicker referenceKey="" value={value} min={min} max={max} onClick={setValue} />
-      </Grid>
-    </Grid>
-  );
-
-  return (
-    <Grid container direction="column" alignItems="center" justify="center">
-      <Grid container item xs>
-        {!disableSetNewButton && (
-          <Button
-            onClick={handleOpen}
-            color="primary"
-            variant="contained"
-            className={classes.openButton}
-          >
-            Set New
-          </Button>
-        )}
-      </Grid>
-      {rotaryEncoder !== null ? (
-        <RotaryEncodeController
-          isActive={true}
-          value={value}
-          onClick={(num: number) => {
-            setValue(num);
-          }}
-          min={min}
-          max={max}
-        />
-      ) : null}
-      <ModalPopup
-        withAction={true}
-        label="Set New"
-        open={open}
-        onClose={handleClose}
-        onConfirm={handleConfirm}
-      >
-        {modalContent}
-      </ModalPopup>
-    </Grid>
-  );
-};
-
-/**
- * SetValueContent
- *
- * @component A container for setting the value of modal settings in multistep popup.
- * This is a modified version of ValueModal
- *
- * Uses the [[ContentProps]] interface
- * @returns JSX.Element
- */
-
-export const SetValueContent = ({
+const ValueModal = ({
```

Comment:
This component still needs a jsdoc code comment explaining it. Maybe you could move the jsdoc comment from the top of the file down to here, and then make the code comment here more detailed while making the code comment at the top of the file more general?